packages/data-point/lib/reducer-path/factory.test.js

```diff
@@ -33,3 +33,116 @@ describe('reducer/reducer-path#create', () => {
     expect(reducer.asCollection).toBe(true)
   })
 })
+
+describe('ReducerPath#body', () => {
+  test('resolve "empty" jsonpath to entire value', () => {
+    const acc = {
+      value: 'test'
+    }
+    const result = factory.create('$').body(acc)
+    expect(result).toBe('test')
+  })
+
+  test('resolve "$." to entire value', () => {
+    const acc = {
+      value: 'test'
+    }
+    const result = factory.create('$.').body(acc)
+    expect(result).toBe('test')
+  })
+
+  test('resolve prefix ".." with valid jsonpath to resolved value', () => {
+    const acc = {
+      value: {
+        a: ['test']
+      },
+      locals: 'test2'
+    }
+    let result = factory.create('$..value.a[0]').body(acc)
+    expect(result).toBe('test')
+    result = factory.create('$..locals').body(acc)
+    expect(result).toBe('test2')
+  })
+
+  test('resolve valid jsonpath to resolved value', () => {
+    const acc = {
+      value: {
+        a: ['test']
+      }
+    }
+    const result = factory.create('$a[0]').body(acc)
+    expect(result).toBe('test')
+  })
+
+  test('resolve valid collection path to resolved value', () => {
+    const acc = {
+      value: [
+        {
+          a: {
+            b: {
+              c: 1
+            }
+          }
+        },
+        {
+          a: {
+            b: {
+              c: 2
+            }
+          }
+        },
+        {
+          a: {
+            b: {
+              c: 3
+            }
+          }
+        }
+      ]
+    }
+    const result = factory.create('$a.b.c[]').body(acc)
+    expect(result).toEqual([1, 2, 3])
+  })
+
+  test('resolve invalid collection path key to array of undefined', () => {
+    const acc = {
+      value: [
+        {
+          a: {
+            b: {
+              c: 1
+            }
+          }
+        },
+        {
+          a: {
+            b: {
+              c: 2
+            }
+          }
+        },
+        {
+          a: {
+            b: {
+              c: 3
+            }
+          }
+        }
+      ]
+    }
+    const result = factory.create('$a.b.d[]').body(acc)
+    expect(result).toEqual([undefined, undefined, undefined])
```

Comment:
this is crazy... but it is what it is... :)

Reply:
Is there a better expected result? This makes sense to me when considering that some of the items in the array might not be undefined.

Here's a rough example:

```javascript
const acc = {
  value: [
    {
      a: {
        b: {
          d: 1
        }
      }
    },
    {
      a: {
        b: {
          c: 2
        }
      }
    },
    {
      a: {
        b: {
          d: 3
        }
      }
    }
  ]
}
const result = factory.create('$a.b.d[]').body(acc)
expect(result).toEqual([1, undefined, 3])
```

Reply:
yeah def, was just pointing out crazy to get a bunch of undefined values :)